fix(@angular-devkit/build-angular): switch to license-checker-webpack-plugin

dgp1130 opened the pull request:
Fixes #16193, #14095.

This replaces the existing `license-webpack-plugin` with the new `license-checker-webpack-plugin`. The former would output multiple `3rdpartylicenses.txt` files which would trigger warnings in Webpack. The later does not emit this warning and still appears to log all third party licenses in a reasonable fashion.

The output format is changed, see https://www.diffchecker.com/puCPV7LQ for a more complete before-and-after comparison of a simple example Angular app (with jQuery and Bootstrap installed).

It is worth discussing if we want to go down this route or if we should stick with `license-webpack-plugin` and attempt to fix it rather than changing the entire dependency.

alan-agius4 commented:
Out of curiosity can you check if it fixes also https://github.com/angular/angular-cli/issues/14095

alan-agius4 commented on packages/angular_devkit/build_angular/package.json:
```diff
@@ -33,13 +33,13 @@
     "less": "3.10.3",
     "less-loader": "5.0.0",
-    "license-webpack-plugin": "2.1.3",
+    "license-checker-webpack-plugin": "^0.1.4",
```
Please pin down

dgp1130 replied:
Done. I left the `@types/...` definition alone since that's just a dev dependency. Should we also lock down that version as well, or is this sufficient?

alan-agius4 replied:
That is sufficient. Thanks

dgp1130 commented:
@alan-agius4, I tried out on their repro case and I do see `open-iconic` in the output. I see it as:

```
--------------------------------------------------------------------------------
open-iconic v1.1.1 - Iconic
git+https://github.com/iconic/open-iconic.git
--------------------------------------------------------------------------------

(MIT License OR SIL OFL 1.1)
```

The user said they expected to see that repo's `FONT-LICENSE` and `ICON-LICENSE`. I don't think those particular files are picked up, as this seems to come from the `package.json`. I think that would be expected as those two `*-LICENSE` names are non-standard. If so, we can probably close that issue as well with this PR.

alan-agius4 commented:
Thanks @dgp1130 for checking the ionic issue. Will update the issue as suggested.

dgp1130 commented:
Discussed this PR, and we'd rather not change a dependency like this so close to the v9 release.

Decision is to instead suppress the warnings for v9, and then for v9.x we can re-evaluate if we want to try and fix the existing dependency or change to this one.

dgp1130 commented:
Now is probably a good time to make this happen, I'll revisit and update this PR.

dgp1130 commented:
Rebased the PR on master and double checked that it still appears to be working as expected. I also added a second commit reverting the log suppression code that is no longer necessary.

I believe this can merge to patch as I don't think we make guarantees about the format of the `3rdpartylicenses.txt` file. Let me know if this should be considered a breaking change and merged only to master for v10.

alan-agius4 commented:
The revert commit message should be reworded to 
```
revert: "fix(@angular-devkit/build-angular): suppress duplicate 3rdpartylicenses.txt warning
```

I think since we are changing the dependency and the output, it's safer to target only master.

dgp1130 commented:
Fixed the commit message, rebased to `master` again, and updated the labels to target `master` only.

alan-agius4 commented on packages/angular_devkit/build_angular/package.json:
```diff
@@ -52,7 +52,6 @@
     "sass-loader": "8.0.2",
     "semver": "7.1.3",
     "source-map": "0.7.3",
-    "source-map-support": "0.5.16",
```
Oops, please add this back as this is a direct depedency of `build-angular`.

dgp1130 replied:
Done. Any particular reason this is needed but not detected? I'm not sure how `yarn` decided to remove this.

alan-agius4 replied:
I am aware of any functionality from yarn to remove dependencies.

dgp1130 replied:
Hmm, strange. I'm not sure why these got removed/reordered then. I think this happened when I did `yarn add`, but I'm not too sure as it was some months ago when I first made the PR.

dgp1130 commented:
I also rebased onto `master`.
